src/javaServerStarter.ts

```diff
@@ -59,6 +60,48 @@ export function runServer(workspacePath, javaConfig): Thenable<StreamInfo> {
 	});
 }
 
+export function runJavaC(sourcePath, outputPath) {
+	return requirements.resolveRequirements().catch(error => {
+		//show error
+		window.showErrorMessage(error.message, error.label).then((selection) => {
+			if (error.label && error.label === selection && error.openUrl) {
+				commands.executeCommand(Commands.OPEN_BROWSER, error.openUrl);
+			}
+		});
+		// rethrow to disrupt the chain.
+		throw error;
+	}).then(requirements => {
+		return new Promise(function(resolve, reject) {
+			let child = path.resolve(requirements.java_home + '/bin/javac');
+		    let params = [sourcePath, '-d', path.resolve(outputPath)];
+
+			console.log('Executing ' + child + ' ' + params.join(' '));
+
+			let childProcess = cp.spawn(child, params, {
+				silent: true,
+			});
+			childProcess.stdout.on('data', function (data) {
+				console.log('stdout: ' + data);
+			});
+			childProcess.stderr.on('data', function (data) {
+				console.log('stderr: ' + data);
+			});
+			childProcess.once('error', function (err) {
+				reject(err);
+			});
+			childProcess.on('exit', function (code) {
+				if (code === 0) {
+					resolve();
+				}
+				else {
+					reject(code);
```

Comment:
reject handler would send the error to vscode panel, rethink about it, i think it is ok to reject twice even if it is hit twice